Applications/FileIO/SWMM/SWMMInterface.cpp

```diff
@@ -0,0 +1,1248 @@
+    if (obj_type == SwmmObject::SUBCATCHMENT)
+    {
+        if (var_idx < n_obj_params[0])
+            return subcatchment_vars[var_idx];
+        if (var_idx < n_obj_params[0]+n_pollutants)
+            return _pollutant_names[var_idx-n_obj_params[0]];
+    }
+    if (obj_type == SwmmObject::NODE)
+    {
+        if (var_idx < n_obj_params[1])
+            return node_vars[var_idx];
+        if (var_idx < n_obj_params[1]+n_pollutants)
+            return _pollutant_names[var_idx-n_obj_params[1]];
+    }
+    if (obj_type == SwmmObject::LINK)
+    {
+        if (var_idx < n_obj_params[2])
+            return link_vars[var_idx];
+        if (var_idx < n_obj_params[2]+n_pollutants)
+            return _pollutant_names[var_idx-n_obj_params[2]];
+    }
+    if (obj_type == SwmmObject::SYSTEM && var_idx < n_obj_params[3])
+    {
+        return system_vars[var_idx];
+    }
+    ERR ("SwmmInterface::getArrayName() - Index error, no name found.");
+    return std::string("");
+}
+
+bool SwmmInterface::addRainGaugeTimeSeriesLocations(std::ifstream &in)
+{
+    std::string line;
+    getline(in, line);
+    while (!isSectionFinished(line))
+    {
+        if (isCommentLine(line))
+        {
+            getline(in, line);
+            continue;
+        }
+
+        std::vector<std::string> const split_str (BaseLib::splitString(line));
+        if (split_str.size() != 8)
+        {
+            ERR ("Rain gauge parameter format not recognised.");
+            return false;
+        }
+
+        for (auto& stn : _rain_gauges)
+        {
+            if (stn.first.getName() == split_str[0] && split_str[4] == "FILE")
+                stn.second = split_str[5].substr(1, split_str[5].size()-2);
+        }
+
+        getline(in, line);
+    }
+
+    for (auto const& stn : _rain_gauges)
+        if (stn.second.empty())
+            WARN ("No associated time series found for rain gauge \"%s\".", stn.first.getName().c_str());
+    return true;
+}
+
+bool SwmmInterface::readPollutants(std::ifstream &in)
+{
+    std::string line;
+    getline(in, line);
+    while (!isSectionFinished(line))
+    {
+        if (isCommentLine(line))
+        {
+            getline(in, line);
+            continue;
+        }
+
+        std::vector<std::string> split_str (BaseLib::splitString(line));
+        if (split_str.size() < 10)
+        {
+            ERR ("Parameter format for pollutants not recognised.");
+            return false;
+        }
+
+        _pollutant_names.push_back(split_str[0]);
+        getline(in, line);
+    }
+    return true;
+}
+
+bool SwmmInterface::isSectionFinished(std::string const& str)
+{
+    if (str.empty())
+        return true;
+
+    std::size_t const pos_beg = str.find_first_not_of(' ', 0);
+    if (pos_beg == str.find_first_of(" \n", pos_beg))
+        return true;
+
+    return false;
+}
+
+bool SwmmInterface::isCommentLine(std::string const& str)
+{
+    return (str.compare(str.find_first_not_of(' ', 0),2,";;") == 0);
```

Comment:
This is checked in the method above. This method is called afterwards when we already know there is *something* written in this line.